src/frontend/js/hooks/useTeacherCoursesSearch/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { useCourseProductUnion } from 'hooks/useCourseProductUnion';
+import { CourseListItem, CourseProductRelation, ProductType } from 'types/Joanie';
+
+const useTeacherCoursesSearch = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [courseAndProductList, setCourseAndProductList] = useState<
+    (CourseListItem | CourseProductRelation)[]
+  >([]);
+  const [isInitialLoading, setIsInitialLoading] = useState(false);
+
+  const { data, isLoading, next, hasMore, count } = useCourseProductUnion({
+    query: searchParams.get('query') || undefined,
```

Comment:
is `|| undefined` needed?

Reply:
It's either that or adding `Nullable` to `useCourseProductUnion` query props.
Both a equivalent.